connect-inject/endpoints_controller.go

```diff
@@ -225,7 +226,25 @@ func (r *EndpointsController) createServiceRegistrations(pod corev1.Pod, service
 	proxyConfig := &api.AgentServiceConnectProxyConfig{
 		DestinationServiceName: serviceName,
 		DestinationServiceID:   serviceID,
-		Config:                 nil, // TODO: add config for metrics (upcoming PR)
+		Config:                 map[string]interface{}{},
+	}
+
+	// If metrics are enabled, the proxyConfig should set envoy_prometheus_bind_addr to a listener on 0.0.0.0 on
+	// the prometheusScrapePort that points to a metrics backend. The backend for this listener will be determined by
+	// the envoy bootstrapping command (consul connect envoy) configuration in the init container. If there is a merged
+	// metrics server, the backend would be that server. If we are not running the merged metrics server, the backend
+	// should just be the Envoy metrics endpoint.
+	enableMetrics, err := r.MetricsConfiguration.enableMetrics(pod)
+	if err != nil {
+		return nil, nil, err
+	}
+	prometheusScrapePort, err := r.MetricsConfiguration.prometheusScrapePort(pod)
```

Comment:
these can be nested under a check for `enableMetrics`:

```go
	if enableMetrics {
		prometheusScrapePort, err := r.MetricsConfiguration.prometheusScrapePort(pod)
		if err != nil {
			return nil, nil, err
		}
		prometheusScrapeListener := fmt.Sprintf("0.0.0.0:%s", prometheusScrapePort)
		proxyConfig.Config["envoy_prometheus_bind_addr"] = prometheusScrapeListener
	}

```

Reply:
nice cleanup!